### PR TITLE
JBIDE-19795: Adding a separate preference for Node. Changing Bower launch logic. Adding Node & Bower detectors. Refactoring

### PR DESCRIPTION
**Pull request by ibuziuk**

### CODE REVIEW

File: plugins/org.jboss.tools.jst.js/src/org/jboss/tools/jst/js/bower/internal/launch/shortcut/BowerUpdate.java [hunk 1]
```diff
@@ -40,24 +36,23 @@ protected String getLaunchName() {
...
-					}
-					return true;
+			if (resource.getType() == IResource.FILE && BowerConstants.BOWER_JSON.equals(resource.getName())) {
+				workingDir = resource.getParent().getFullPath().toOSString();
+			} else if (resource.getType() == IResource.FOLDER) {
+				workingDir = resource.getFullPath().toOSString();
+			} else if (resource.getType() == IResource.PROJECT) {
+				IProject project = (IProject) resource;
+				IFile file = project.getFile(BowerConstants.BOWER_JSON);
+				if (file != null && file.exists()) {
```
Comment (alexeykazakov) [inline plugins/org.jboss.tools.jst.js/src/org/jboss/tools/jst/js/bower/internal/launch/shortcut/BowerUpdate.java]: file is never null here

**Comment by alexeykazakov** [comment]
Isn't it time to add some tests? :)

**Comment by alexeykazakov** [comment]
Rebased and pushed to master.

**Comment by ibuziuk** [comment]
@alexeykazakov agree it's high time for tests. Will add some in the nearest future ;-)

**Pull request closed without merging**